command/agent/auth/kubernetes/kubernetes.go

```diff
@@ -54,14 +74,15 @@ func NewKubernetesAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 
 func (k *kubernetesMethod) Authenticate(ctx context.Context, client *api.Client) (string, map[string]interface{}, error) {
 	k.logger.Trace("beginning authentication")
-	content, err := ioutil.ReadFile(serviceAccountFile)
+
+	jwtString, err := k.readJWT()
 	if err != nil {
-		log.Fatal(err)
+		return "", nil, errwrap.Wrapf("error reading JWT with Kubernetes Auth: {{err}}", err)
```

Comment:
This needs good review - the previous behavior was to log the error and exit completely, which seemed to be excessive.

Reply:
That was almost certainly something accidentally left in from debug sessions when it was initially being written, so this looks good.